src/fp6.rs

```diff
@@ -56,11 +56,46 @@ pub struct Fp6 {
 
 impl fmt::Debug for Fp6 {
     fn fmt(&self, f: &mut Formatter) -> fmt::Result {
-        write!(
-            f,
-            "{:?} + {:?}*u + {:?}*u^2 + {:?}*u^3 + {:?}*u^4 + {:?}*u^5",
-            self.c0, self.c1, self.c2, self.c3, self.c4, self.c5
-        )
+        let coeffs: [Fp; 6] = self.into();
+
+        let format_term = |coef: Fp, degree: usize| -> String {
+            if coef == Fp::one() && degree > 0 {
+                let exp = if degree > 1 {
+                    format!("^{}", degree)
+                } else {
+                    "".to_string()
+                };
+                format!("u{}", exp)
+            } else {
+                let exp =  if degree > 0 {
+                    format!("u{}", if degree > 1 { format!("^{}", degree) } else { "".to_string() })
+                } else {
+                    "".to_string()
+                };
+                format!("{}{}", coef, exp)
+            }
+        };
+
+        let elem_rep = coeffs
+            .iter()
+            .enumerate()
+            .filter_map(|(i, &coef)| {
+                if coef == Fp::zero() {
+                    None
+                } else {
+                    Some(format_term(coef, i))
+                }
+            })
+            .collect::<Vec<String>>()
+            .join(" + ");
+
+        if elem_rep.is_empty() {
+            write!(f, "0")?; // Handle the case where all coefficients are zero
+        } else {
+            write!(f, "{}", elem_rep)?;
+        }
+
+        Ok(())
```

Comment:
Same comment on special case handling than `Fp3`. And `Fp6` does implement `is_zero()` directly, which can then be converting to a `bool`.